File: UI/ViewModels/QueryViewModel.cs
using System;
using System.Windows.Input;
using UI.Models;

namespace UI.ViewModels {

    public class QueryViewModel {

        public QueryModel Model { get; set; }

        public ICommand Connect { get; set; }

        public QueryViewModel() {
            Model = new QueryModel();

            Connect = new RelayCommand(
                x => !String.IsNullOrWhiteSpace(Model.Host) &&
                     !String.IsNullOrWhiteSpace(Model.Channel) &&
                     !String.IsNullOrWhiteSpace(Model.TargetPath) &&
                     Model.Target != null,
                x => Model.Connect());
        }
    }
}

File: UI/ViewModels/QueryViewModel.fs
namespace UI.ViewModels

open System
open System.Windows.Input

open UI.Commands
open UI.Models

type QueryViewModel() =

    let model = new QueryModel()

    let connectCommand =
        new RelayCommand(
            (fun _ -> not (String.IsNullOrWhiteSpace(model.Host)) &&
                      not (String.IsNullOrWhiteSpace(model.Channel)) &&
                      not (String.IsNullOrWhiteSpace(model.TargetPath))),
            (fun _ -> model.Connect()))

    member public x.Model = model
    member public x.Connect = connectCommand